modules/govuk/spec/classes/govuk_nodes_spec_optional.rb

```diff
@@ -53,16 +54,41 @@
 
     let(:hiera_config) { temporary_hiera_file.path }
 
-    # Pull in some required bits from top-level site.pp
-    let(:pre_condition) do
-      <<-EOT
-        $govuk_node_class = "#{class_name}"
+    if class_name == "ci_agent"
+      # Pull in some required bits from top-level site.pp
+      let(:pre_condition) do
+        <<-EOT
+          $govuk_node_class = "#{class_name}"
+
+          $lv = hiera('lv',{})
+          create_resources('govuk_lvm', $lv)
+          $mount = hiera('mount', {
+            '/var/lib/jenkins' => {
+              'disk' => '/dev/mapper/jenkins-data',
+              'govuk_lvm' => 'data',
+              'mountoptions' => 'defaults',
+            },
+            '/var/lib/docker' => {
+              'disk' => '/dev/mapper/data-docker',
+              'govuk_lvm' => 'docker',
+              'mountoptions' => 'defaults',
+            },
+          })
+          create_resources('govuk_mount', $mount)
+        EOT
+      end
+    else
```

Comment:
The copy/paste makes it quite hard to see what this code is doing at a glance. Can we factor out the part that's the same in both cases of the conditional?

The [docs](https://rspec-puppet.com/documentation/classes/#specifying-code-to-include-before) say that pre_condition can be an array of strings, so if I were writing this I would assign the common stuff (i.e. everything except `$mount = ...`) to one string, and the `$mount = ...` part (that depends on whether it's a CI node) to another string and then write something like `let(:pre_condition) { [$lvm_precondition, $mount_precondition] }`.

I'd also add a comment to say why the extra stuff is needed for the CI node classes (as it's completely opaque to the reader otherwise).

Reply:
Thanks. We could simplify this a bit rather than this quick stab at things, let me have a look.